packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -813,6 +813,276 @@ class _TextSelectionHandleOverlayState
   }
 }
 
+/// An interface to receive selection handlers from [TextSelectionGestureDetectorBuilder].
+///
+/// Class implements [TextSelectionGestureDetectorBuilderClient] will be able to use
+/// [TextSelectionGestureDetectorBuilder].
+///
+/// See also:
+///
+///  * [TextSelectionGestureDetectorBuilder], which provide basic selection handlers
+///    for [EditableText].
+///  * [TextField], which implements this abstract class
+///  * [CupertinoTextField], which implements this abstract class
+abstract class TextSelectionGestureDetectorBuilderClient {
+  /// Key to underlying [EditableText].
+  ///
+  /// This property will be used in retrieving [EditableTextState] and [RenderEditable]
+  /// for triggering certain selection event.
+  GlobalKey<EditableTextState> get editableTextKey;
+
+  /// Whether force press is enabled.
+  ///
+  /// If true, [TextSelectionGestureDetectorBuilder] will provide handlers for
+  /// force press selection.
+  bool get forcePressEnabled;
+
+  /// Whether selection is enabled.
+  ///
+  /// If false, [TextSelectionGestureDetectorBuilder] will disable certain
+  /// selection handlers
+  bool get selectionEnabled;
+}
+
+/// Builds a [TextSelectionGestureDetector] to wrap an [EditableText].
+///
+/// The class implements sensible defaults for many user interactions
+/// with an [EditableText] (see the documentation of the various gesture handler
+/// methods, e.g. [onTapDown], [onFrocePress], etc.). Subclasses of
+/// [EditableTextSelectionHandlesProvider] can change the behavior performed in
+/// responds to these gesture events by overriding the corresponding handler
+/// methods of this class.
+///
+/// The resulting [TextSelectionGestureDetector] to wrap an [EditableText] is
+/// obtained by calling [buildGestureDetector].
+///
+/// See also:
+///
+///  * [TextField], which uses a subclass to implement the Material-specific
+///    gesture logic of an [EditableText].
+///
+///  * [CupertinoTextField], which uses a subclass to implement the
+///    Cupertino-specific gesture logic of an [EditableText].
+class TextSelectionGestureDetectorBuilder {
+  /// Create a [TextSelectionGestureDetectorBuilder] that provide selection
+  /// handlers to input client.
+  TextSelectionGestureDetectorBuilder({
+    @required this.client,
+  }) : assert(client != null);
+
+  /// The client to be provided a handler for.
```

Comment:
The delegate for this [TextSelectionGestureDetectorBuilder].

The delegate provides the builder with information about what actions can currently be performed on the textfield. Based on this, the builder adds the correct gesture handlers to the gesture detector.